apps/users/views.py

```diff
@@ -231,6 +232,78 @@ def edit(request):
 
 @write
 @login_required
+def t_shirt(request):
+    if not waffle.switch_is_active('t-shirt-orders'):
+        raise http.Http404()
+
+    MIN_PERSONA_ADU = 10000
+
+    user = UserProfile.objects.get(pk=request.user.id)
+    try:
+        instance = TShirtOrder.objects.get(user=user)
+    except TShirtOrder.DoesNotExist:
+        instance = None
+
+    eligible = (
+        bool(instance) or
+
+        AddonUser.objects.filter(
+            user=user,
+            role__in=(amo.AUTHOR_ROLE_OWNER, amo.AUTHOR_ROLE_DEV),
+            addon__type=amo.ADDON_EXTENSION,
+            addon__disabled_by_user=False)
+        .filter(
+            Q(addon__is_listed=True,
+              addon___current_version__files__status__in=amo.REVIEWED_STATUSES,
+              addon__status__in=amo.REVIEWED_STATUSES) |
+            Q(addon__is_listed=False,
+              addon__versions__files__is_signed=True))
+        .exists() or
+
+        Addon.objects.filter(
+            authors=user,
+            type=amo.ADDON_PERSONA,
+            status=amo.STATUS_PUBLIC,
+            disabled_by_user=False)
+        .aggregate(Sum('average_daily_users')) > MIN_PERSONA_ADU)
```

Comment:
You're comparing a dict to an integer. In my test, it's comparing `{'average_daily_users__sum': None} > 10000` which returns True, for some odd reason. So everybody is always be eligible.

I think tests for this "eligible" test would be welcome ;) Maybe make `eligible` a helper function? That would be easier to test?